src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -478,16 +478,18 @@ public virtual bool IsImplicitClass
         /// </summary>
         public abstract override string Name { get; }
 
+#nullable enable
         /// <summary>
         /// Return the name including the metadata arity suffix.
         /// </summary>
         public override string MetadataName
         {
             get
             {
-                return MangleName ? MetadataHelpers.ComposeAritySuffixedMetadataName(Name, Arity) : Name;
```

Comment:
>MangleName [](http://example.com/codeflow?start=23&length=10)

It is not clear why this property is ignored.

Reply:
`MangleName` is really just a check for whether `Arity > 0` here. `ComposeAritySuffixedMetadataName` will already return the name without a suffix when `Arity == 0`, so it's simpler to just eliminate the check here and build/return the appropriate name based on the arity and associated file.

Reply:
I was wrong about this :) this needs to be restored to ensure we produce the correct metadata name in some cases, particularly for synthesized types.